src/Microsoft.Extensions.FileProviders.Physical/ConcurrentLruCache.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.FileProviders.Physical
+{
+    /// <summary>
+    /// Cache with a fixed size that evicts the least recently used members.
+    /// Thread-safe.
+    /// </summary>
+    internal class ConcurrentLruCache<K, V>
+    {
+        private readonly int _capacity;
+
+        private struct CacheValue
+        {
+            public V Value;
+            public LinkedListNode<K> Node;
+        }
+
+        private readonly Dictionary<K, CacheValue> _cache;
+        private readonly LinkedList<K> _nodeList;
```

Comment:
Just some remarks about `_nodeList`, maybe stupid. Sorry if it is the case...
I would remove `_nodeList`:
- The `LinkedListNode` will create memory fragmentation.
- Removing the older nodes is not a high priority task, it can be delegated to a background task, like in the Memory Cache.
- It will save time because the `GetOrAdd()` won't have to move the node to top each time.
- I'm not sure, but if you do this you may remove `_lockObject` too, and keep only a high performance `ConcurrentDictionary` in `_cache`. It would create some collisions in rare cases (two requests want to add the same value to the cache), but I don't think that its a problem.